### PR TITLE
Changing is saved

### PR DESCRIPTION
**Pull request by AvitalGitelman**

### CODE REVIEW

**Review by edenshaul — COMMENTED** [review]
you need to do a rebase from the main branch

File: eatin-client/src/components/accounts/profile.tsx [hunk 1]
```diff
@@ -13,27 +13,28 @@ import TabList from "@mui/joy/TabList";
...
-import { useGetSavedRecipes } from "../functions/useGetSavedRecipes";
+import { useGetRecipesConnectionIsSaved } from "../../graphql/queries/recipes_connection_is_saved.query";
 
 export const Profile: FC = () => {
     const [categoryFilter, setCategoryFilter] = useState("");
     const [cuisineFilter, setCuisineFilter] = useState("");
     const { currentUser } = useAuth();
     const { data, error, loading } = useGetUserByIdQuery({
-        variables: { id: !!currentUser?.uid ? currentUser?.uid : "" },
+        variables: { id: currentUser ? currentUser.uid : "" },
```
Comment (edenshaul) [inline eatin-client/src/components/accounts/profile.tsx:23]: In this file and the rest files where you get some parameter of `currentUser`, add a `useMemo` hook that will calculate the parameter once. like this: \
`const kaki = useMeme(() => currentUser ? currentUser.uid : "", [currentUser])`

File: eatin-client/src/components/ui/RecipeItem.tsx [hunk 2]
```diff
@@ -4,17 +4,21 @@ import { FC } from "react";
...
+import { useAuth } from "../../context/auth-context";
 
 type Props = {
     recipe: Recipe;
-    updateSavedRecipes: (isSaved: boolean, recipeIndex: number) => void;
+    updateSavedRecipes: (recipeIndex: number) => void;
 };
 
 export const RecipeItem: FC<Props> = ({ recipe, updateSavedRecipes }) => {
+    const { currentUser } = useAuth();
```
Comment (edenshaul) [inline eatin-client/src/components/ui/RecipeItem.tsx:15]: remove, you don't use this

File: eatin-client/src/components/ui/RecipeItemOverlay.tsx [hunk 3]
```diff
@@ -31,10 +31,10 @@ export const RecipeItemOverlay: FC<Props> = ({
                     <CancelIcon className="delete-from-list-icon p-0" />
                 </Button>
                 <BookmarkButton
-                    onChange={(value) => updateSavedRecipes(value, index)}
+                    onChange={() => updateSavedRecipes(index)}
                     recipeID={index}
                     recipeName={recipe_title}
-                    isClicked={is_saved ? true : false}
+                    isClicked={is_saved ? is_saved : false}
```
Comment (edenshaul) [inline eatin-client/src/components/ui/RecipeItemOverlay.tsx:37]: you can do this by just checking if `is_saved` has a value

File: eatin-client/src/graphql/queries/recipes_connection_is_saved.query.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            });
+            setData(newData);
+            setLoading(false);
+        } catch (e) {
+            console.log(`Error has occured in recipes_connection_is_saved query - ${e}`);
+            setError(e);
+        }
+    }, [isSaved, userId]);
+
+    // const updateIsSaved = useCallback(
```
Comment (edenshaul) [inline eatin-client/src/graphql/queries/recipes_connection_is_saved.query.ts:56]: delete

File: eatin-client/src/graphql/queries/sections.query.ts [hunk 5]
```diff
@@ -39,6 +39,9 @@ export function useGetSections(userId: string) {
                                       total_time
                                       url
                                       vote_count
+                                        is_saved
```
Comment (edenshaul) [inline eatin-client/src/graphql/queries/sections.query.ts:42]: formetter here

File: eatin-client/src/graphql/queries/userRecipes.query.ts [hunk 6]
```diff
@@ -46,48 +54,102 @@ export const GET_USERRECIPES_BY_USER_AND_IS_SAVED = gql`
...
-                difficulty
-                total_time
-            }
+            is_uploaded
+            given_comment
         }
     }
 `;
+
+// export const GET_USERRECIPES_BY_USER_WITH_RECIPE = gql`
```
Comment (edenshaul) [inline eatin-client/src/graphql/queries/userRecipes.query.ts:123]: delete

File: eatin-server/migrate/V8__add_comment_and_is_upload_to_userrecipes.pgsql [hunk 7]
```diff
@@ -0,0 +1,2 @@
+alter table userrecipes add column is_uploaded boolean not null default false;
+alter table userrecipes add column given_comment varchar(100) not null default '';
```
Comment (edenshaul) [inline eatin-server/migrate/V8__add_comment_and_is_upload_to_userrecipes.pgsql:2]: you should change this column type from `varchar(100)` to `text` so we will be able to save comments without worrying about text size

File: eatin-server/src/graphql/recipes/recipes.service.ts [hunk 8]
```diff
@@ -1,60 +1,201 @@
...
+      .leftJoin(
+        Userrecipes,
+        'userrecipes',
+        `recipes.index = userrecipes.recipe_index and userrecipes.user_id = '${userID}'`,
+      )
+      .orderBy('recipes.rating', 'DESC')
+      .where('recipes.category = :category', { category })
+      .take(20)
+      .getRawMany();
+  }
```
Comment (edenshaul) [inline eatin-server/src/graphql/recipes/recipes.service.ts:154]: very nice

File: eatin-server/src/graphql/userRecipes/userRecipes.dto.ts [hunk 9]
```diff
@@ -4,5 +4,7 @@ export interface UserrecipeDTO {
   user_id: string;
   recipe_index: number;
   is_saved: boolean;
-  recipe?: Recipes;
+  is_uploaded: boolean;
+  given_comment: string;
+  // recipe?: Recipes;
```
Comment (edenshaul) [inline eatin-server/src/graphql/userRecipes/userRecipes.dto.ts:9]: delete

File: eatin-server/src/graphql/userRecipes/userRecipes.model.ts [hunk 10]
```diff
@@ -14,7 +21,16 @@ export class Userrecipes {
...
   @Field()
-  @OneToOne(() => Recipes)
-  @JoinColumn({ name: 'recipe_index' })
-  recipe: Recipes;
+  @Column('boolean', { nullable: false })
+  is_uploaded: boolean;
+  @Field()
+  @Column('varchar', { length: 100, nullable: false })
+  given_comment: string;
+  // @OneToOne(() => Recipes)
```
Comment (edenshaul) [inline eatin-server/src/graphql/userRecipes/userRecipes.model.ts:29]: delete

File: eatin-server/src/graphql/userRecipes/userRecipes.resolver.ts [hunk 11]
```diff
@@ -43,21 +43,59 @@ export class UserrecipeResolver {
...
+
+  @Query((returns) => Userrecipes)
+  async userRecipesByUserAndRecipe(
+    @Args('userID') userID: string,
+    @Args('recipeID') recipeID: number,
+  ): Promise<Userrecipes> {
+    return await this.userRecipesService.findByUserAndRecipe(userID, recipeID);
+  }
+
+  // @Query((returns) => [Userrecipes])
```
Comment (edenshaul) [inline eatin-server/src/graphql/userRecipes/userRecipes.resolver.ts:92]: delete

File: eatin-server/src/graphql/userRecipes/userRecipes.service.ts [hunk 12]
```diff
@@ -45,6 +45,24 @@ export class UserrecipesService {
...
+    userID: string,
+    isUploaded: boolean,
+  ): Promise<Userrecipes[]> {
+    return this.userRecipesRepository.findBy({
+      user_id: userID,
+      is_uploaded: isUploaded,
+    });
+  }
+
+  findByUserAndisCommentExists(userID: string): Promise<Userrecipes[]> {
```
Comment (edenshaul) [inline eatin-server/src/graphql/userRecipes/userRecipes.service.ts:58]: make the letter i uppercase

File: eatin-server/src/graphql/userRecipes/userRecipes.service.ts [hunk 13]
```diff
@@ -71,14 +107,40 @@ export class UserrecipesService {
...
       .createQueryBuilder('userrecipe')
-      .leftJoinAndSelect('userrecipe.recipe', 'recipe')
+      .where('userrecipe.recipe_index = :recipeID', { recipeID })
       .where('userrecipe.user_id = :userID', { userID })
-      .getMany();
+      .where('length(given_comment)>0')
+      .getOne();
   }
 
+  // async findByUserWithRecipe(userID: string): Promise<Userrecipes[]> {
```
Comment (edenshaul) [inline eatin-server/src/graphql/userRecipes/userRecipes.service.ts:136]: delete

File: eatin-server/migrate/V8__add_comment_and_is_upload_to_userrecipes.pgsql (hunk 7 above)
Comment (AvitalGitelman) [reply eatin-server/migrate/V8__add_comment_and_is_upload_to_userrecipes.pgsql:2]: I need to add another migrate file? cuz i cant change it just as it is

**Pull request merged**